src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java

```diff
@@ -98,22 +99,22 @@ public DataSample getSample(DataSeries dataSeries, int rows) {
         ) {
             parser = csvFormat.parse(input);
 
-            List<List<String>> sampleRows = StreamSupport.stream(parser.spliterator(), false)
+            List<Column> columns = dataSeries.getColumns();
+
+            List<CSVRecord> records = StreamSupport
+                    .stream(parser.spliterator(), false)
                     .limit(rows)
-                    .map(record -> IntStream.range(0, record.size())
-                            .boxed()
-                            .map(record::get)
-                            .collect(Collectors.toList())
-                    )
                     .collect(Collectors.toList());
 
-            List<Column> columns = dataSeries.getColumns();
-
             if (columns.isEmpty()) {
                 // Auto-detect types from CSVRecords
-                Map<String, String> csvMap = parser.iterator().next().toMap(); // Map<Header, Data>
 
                 List<String> headers = csvSeriesConfig.getHeaders();
+                Map<String, String> csvMap = records
+                        .stream()
+                        .findFirst()
+                        .orElseThrow(IllegalStateException::new)
```

Comment:
Please use `InputException` from `fi.jubic.quanta.exception` with a descriptive error message. Something like "Cannot import an empty CSV file".

Reply:
Changed to `InputException`